### PR TITLE
Added Promise Chains

### PR DESCRIPTION
**Pull request by e-schultz**
* added examples on promise chains
* combining promises
* promise.all

Todo:
- [x] Errors

### CODE REVIEW

File: decks/promises/01-promise-chain.mdx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+```javascript
+let promiseA = someRequest('A');
+let promiseB = someRequest('B');
+let PromiseC = someRequest('C');
+
+Promise.all([promiseA,promiseB,promiseC])
+    .then(results=>{
+        console.log('Hello Results',results)
+    })
+```
```
Comment (renvrant) [inline decks/promises/01-promise-chain.mdx]: This block should be separated as it runs off the screen a little.
I think you should also note here that this usually needs to be polyfilled (we recommend Bluebird)

File: decks/promises/01-promise-chain.mdx [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+* Waits for all promises to resolve
+* Gets passed into the handler as an array
+
+```javascript
+let promiseA = someRequest('A');
+let promiseB = someRequest('B');
+let PromiseC = someRequest('C');
+
+Promise.all([promiseA,promiseB,promiseC])
```
Comment (renvrant) [inline decks/promises/01-promise-chain.mdx]: Here, adding spaces would make this array easier to read I think

File: decks/promises/01-promise-chain.mdx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    .then(results=>{
```
Comment (renvrant) [inline decks/promises/01-promise-chain.mdx]: Again, adding spaces would be helpful

File: decks/promises/01-promise-chain.mdx [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    })
+```
+
+---
+## Promise.all - can also use destructuring
+
+* Waits for all promises to resolve
+* Gets passed into the handler as an array
+
+```javascript
```
Comment (renvrant) [inline decks/promises/01-promise-chain.mdx]: This also runs off the page in the slides

File: decks/promises/01-promise-chain.mdx [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+let promiseA = someRequest('A');
+let promiseB = someRequest('B');
+let PromiseC = someRequest('C');
+
+Promise.all([promiseA,promiseB,promiseC])
```
Comment (renvrant) [inline decks/promises/01-promise-chain.mdx]: Spacing here please

File: decks/promises/01-promise-chain.mdx [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+    })
+```
+
+---
+## API Consideration 
+
+<a href="https://js-ts-training.now.sh/" target="_blank">Simple API</a>
+---
+
+## Exersice Time:
```
Comment (renvrant) [inline decks/promises/01-promise-chain.mdx]: Exercise

File: decks/promises/01-promise-chain.mdx [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+* Get a game by ID
+* Get a list of Platforms 
+* Create a game object with the game name, and the platform name
+
+```js
+let expected = {
+    id: 3,
+    name: "Dragon Quest XI: Echoes of an Elusive Age",
+    platformId: 1,
+    platFormName: "PS4"
```
Comment (renvrant) [inline decks/promises/01-promise-chain.mdx]: `platform`?

File: decks/promises/01-promise-chain.mdx [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+<a href="https://js-ts-training.now.sh/" target="_blank">Simple API</a>
+---
+
+## Exersice Time: 
+
+* Get a game by ID
+* Get a list of Platforms 
+* Create a game object with the game name, and the platform name
+
+```js
```
Comment (renvrant) [inline decks/promises/01-promise-chain.mdx]: This example runs off the page too

**Comment by renvrant** [comment]
Oh, one thing I forgot - please do add bluebird to the RESOURCES.md

File: decks/promises/01-promise-chain.mdx (hunk 1 above)
Comment (e-schultz) [reply decks/promises/01-promise-chain.mdx]: cleaned up the spacing - will mention the polyfill (so used to tooling adding this for us that I don't think about it)

File: decks/promises/01-promise-chain.mdx (hunk 8 above)
Comment (e-schultz) [reply decks/promises/01-promise-chain.mdx]: cleaned up spacing - and condensed the output to one line

File: decks/promises/01-promise-chain.mdx (hunk 7 above)
Comment (e-schultz) [reply decks/promises/01-promise-chain.mdx]: fixed

**Pull request merged**